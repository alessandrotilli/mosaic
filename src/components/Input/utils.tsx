import React from "react";
import { InputAdornment as MUIInputAdornment } from "@material-ui/core";

import { IInputAdornment } from "../../types/Input";
import IconButton from "../IconButton";

export const getAdornment = (adornment?: IInputAdornment) => {
  if (!adornment) {
    return;
  }

  const { icon, onClick } = adornment;
  return (
    <MUIInputAdornment position="end">
      <IconButton disabled={!onClick} icon={icon} onClick={onClick!} size="small" />
    </MUIInputAdornment>
  );
};
