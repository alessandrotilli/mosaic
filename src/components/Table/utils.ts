export const COLUMN_CHECKBOX_PATH = "checkbox-selection";
export const COLUMN_ROW_ACTIONS_PATH = "row-actions";

export const TOOLBAR_HEIGHT = 64;
export const TOOLBAR_HEIGHT_MOBILE = 56;

export const PAGINATION_TOOLBAR_HEIGHT = 64;
