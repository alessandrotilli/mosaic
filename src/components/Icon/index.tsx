import React, { cloneElement, FC, PropsWithChildren, ReactElement, useCallback, useMemo } from "react";
import { ClassNames } from "@emotion/react";
import { keyframes, Skeleton as MUISkeleton } from "@mui/material";

import { IIcon, IIconDimensions, IRenderedIcon } from "../../types/Icon";
import { logWarn } from "../../utils/logger";
import Adornment from "../Adornment";

import { iconsCatalog } from "./utils";

export const DATA_CY_DEFAULT = "icon";

const ICON_DIMENSIONS: IIconDimensions = {
  small: 20,
  medium: 24,
  large: 35,
};

const Icon: FC<PropsWithChildren<IIcon>> = ({
  badge,
  children,
  dataCy = DATA_CY_DEFAULT,
  forwarded = {},
  loading = false,
  name,
  rotate = false,
  size = "medium",
  style: externalStyle,
  tooltip,
}) => {
  const dimensions = useMemo(() => {
    const dimension = ICON_DIMENSIONS[size];
    return { height: dimension, width: dimension };
  }, [size]);

  const props = useMemo(() => {
    let props: IRenderedIcon = {
      ...forwarded,
      "data-cy": dataCy,
      fontSize: size,
    };

    if (externalStyle) {
      props = { ...props, style: { ...externalStyle } };
    }

    if (children) {
      props = { ...props, style: { ...props.style, ...dimensions } };
    }

    return props;
  }, [children, dataCy, dimensions, externalStyle, forwarded, size]);

  const rotateKeyframes = useMemo(
    () => keyframes`
      from: {
        transform: "rotate(0deg)";
      }
      to: {
        transform: "rotate(360deg)";
      }
  `,
    []
  );

  const renderWithAnimation = useCallback(
    (element: any) => {
      return (
        <ClassNames>
          {({ css }) => {
            let animation = "none";
            if (rotate) {
              animation = css`
                ${rotateKeyframes} 2s linear infinite
              `;
            }

            let className = css`
              animation: ${animation};
            `;

            if (forwarded.className) {
              className = forwarded.className;
            }

            return (
              <Adornment badge={badge} tooltip={tooltip}>
                {cloneElement(element as ReactElement<any>, { ...props, className })}
              </Adornment>
            );
          }}
        </ClassNames>
      );
    },
    [badge, forwarded, props, rotate, rotateKeyframes, tooltip]
  );

  if (loading) {
    return <MUISkeleton variant="rectangular" style={{ ...dimensions }} />;
  }

  if (children) {
    return renderWithAnimation(children);
  }

  if (!name) {
    logWarn("Icon", "Skip rendering, both children and name are not set");
    return null;
  }

  const icon = iconsCatalog[name];
  return renderWithAnimation(icon);
};

export default Icon;
