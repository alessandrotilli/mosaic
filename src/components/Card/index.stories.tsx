import React from "react";
import { action } from "@storybook/addon-actions";
import { boolean, select, text } from "@storybook/addon-knobs";

import { Icons } from "../../types/Icon";
import { getAllComposedDataCy } from "../../utils";
import IntlProviderMock, { LocaleMock, MessageMock } from "../../utils/mocks/IntlProviderMock";
import { getDocumentationPage, StoriesWrapper } from "../../utils/stories";
import Button from "../Button";
import IconButton from "../IconButton";
import Typography from "../Typography";

import Card, { CardWithProps, DATA_CY_DEFAULT, DATA_CY_SHORTCUT, LOCALIZABLE_PROPS, SUBPARTS_MAP } from ".";

export default {
  title: "Card",
  component: CardWithProps,
  parameters: {
    ...getDocumentationPage({
      basedOn: "@material-ui/core/Card",
      component: "Card",
      e2eTestInfo: {
        dataCyDefault: DATA_CY_DEFAULT,
        dataCyShortcut: DATA_CY_SHORTCUT,
        subpartsSuffixes: getAllComposedDataCy(SUBPARTS_MAP),
      },
      localizableProps: LOCALIZABLE_PROPS,
    }),
  },
};

export const Canvas = () => (
  <Card
    actions={[
      <Button label="Share" onClick={action("Share Button")} />,
      <IconButton icon={Icons.add} onClick={action("Add action")} />,
    ]}
    collapsible={
      <Typography>
        Traditional mosaics are made of small, flat, roughly square pieces of stone or glass of different colors, known
        as tesserae. Some floor mosaics are made of small rounded pieces of stone and called pebble mosaics. Mosaic
        skinning (covering objects with mosaic glass) is done with thin enameled glass and opaque stained glass. Modern
        mosaic art is made from any material in any size ranging from carved stone, bottle caps, and found objects.
      </Typography>
    }
    dataCy={text("dataCy", DATA_CY_DEFAULT)}
    icon={select("icon", Icons, Icons.add)}
    loading={boolean("loading", false)}
    localized={boolean("localized", false)}
    subtitle={text("subTitle", "Mosaic Card Subtitle")}
    title={text("title", "Mosaic Card")}
    unmountCollapsible={boolean("unmountCollapsible", false)}
  >
    <Typography>
      A mosaic is an artistic picture or design made out of any materials assembled together. Mosaic are used as
      decoration. Architects use mosaic murals for kitchen backsplash, shower wall and entry floor art. Mosaic Craft
      items are used as home decor. Cities often decorate public places such as parks with mosaic murals and sculptures.
    </Typography>
  </Card>
);

export const Basic = () => (
  <StoriesWrapper>
    <Card title="Mosaic">
      <Typography>A mosaic is an artistic picture or design made out of any materials assembled together.</Typography>
    </Card>
  </StoriesWrapper>
);

export const Collapsible = () => (
  <StoriesWrapper>
    <Card
      actions={[<Button icon={{ name: Icons.open_new }} label="Discover" onClick={action("Discover")} />]}
      collapsible={<Typography children="Discover our catalog!" />}
      icon={Icons.business}
      subtitle="The best in town for components"
      title="Mosaic Shop"
    >
      <Typography>Opening hours: 9AM - 6PM</Typography>
    </Card>
  </StoriesWrapper>
);

export const Loading = () => (
  <StoriesWrapper>
    <Card icon={Icons.add} loading title="Mosaic">
      <Typography>A mosaic is an artistic picture or design made out of any materials assembled together.</Typography>
    </Card>
  </StoriesWrapper>
);

export const Localized = () => (
  // IntlProviderMock simulates external IntlProvider context
  <StoriesWrapper>
    <IntlProviderMock locale={select("locale", LocaleMock, LocaleMock.en)}>
      <Card icon={Icons.business} localized title={MessageMock.title} subtitle={MessageMock.subtitle} />
    </IntlProviderMock>
  </StoriesWrapper>
);

export const WithAvatar = () => (
  <StoriesWrapper>
    <Card icon={Icons.business} subtitle="The best in town for components" title="Mosaic Shop">
      <Typography>Opening hours: 9AM - 6PM</Typography>
    </Card>
  </StoriesWrapper>
);

export const WithActions = () => (
  <StoriesWrapper>
    <Card
      actions={[<Button icon={{ name: Icons.open_new }} label="Discover" onClick={action("Discover")} />]}
      icon={Icons.business}
      subtitle="The best in town for components"
      title="Mosaic Shop"
    >
      <Typography>Opening hours: 9AM - 6PM</Typography>
    </Card>
  </StoriesWrapper>
);
