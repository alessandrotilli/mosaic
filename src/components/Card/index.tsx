import React, { cloneElement, FC, Fragment, useMemo, useState } from "react";
import {
  Card as MUICard,
  CardActions as MUICardActions,
  CardContent as MUICardContent,
  CardHeader as MUICardHeader,
  Collapse as MUICollapse,
  useTheme,
} from "@material-ui/core";
import { Skeleton as MUISkeleton } from "@material-ui/lab";

import { ICard } from "../../types/Card";
import { Icons } from "../../types/Icon";
import { getComposedDataCy } from "../../utils";
import localized, { ILocalizableProperty } from "../../utils/hocs/localized";
import Avatar from "../Avatar";
import IconButton from "../IconButton";
import Typography from "../Typography";

export const DATA_CY_DEFAULT = "card";
export const DATA_CY_SHORTCUT = "title";
export const LOCALIZABLE_PROPS: ILocalizableProperty[] = [
  { name: "title", type: "string" },
  { name: "subtitle", type: "string" },
];

export const SUBPARTS_MAP = {
  avatar: {
    label: "Avatar",
  },
  title: {
    label: "Title",
  },
  subtitle: {
    label: "Subtitle",
  },
  content: {
    label: "Content",
  },
  collapse: {
    label: "Collapse",
  },
  collapsibleContent: {
    label: "Collapsible Content",
  },
};

const Card: FC<ICard> = ({
  actions = [],
  children,
  collapsible,
  dataCy = DATA_CY_DEFAULT,
  icon,
  loading = false,
  style,
  subtitle,
  title,
  unmountCollapsible = false,
}) => {
  const [expanded, setExpanded] = useState(false);
  const theme = useTheme();

  const cardHeader = useMemo(
    () => (
      <MUICardHeader
        avatar={
          icon && <Avatar dataCy={getComposedDataCy(dataCy, SUBPARTS_MAP.avatar)} icon={icon} loading={loading} />
        }
        disableTypography
        title={
          <Typography
            bottomSpacing={false}
            dataCy={getComposedDataCy(dataCy, SUBPARTS_MAP.title)}
            loading={loading}
            truncated
            variant="title"
          >
            {title}
          </Typography>
        }
        subheader={
          <Typography
            bottomSpacing={false}
            dataCy={getComposedDataCy(dataCy, SUBPARTS_MAP.subtitle)}
            loading={loading}
            truncated
            variant="caption"
          >
            {subtitle}
          </Typography>
        }
      />
    ),
    [dataCy, icon, loading, subtitle, title]
  );

  return (
    <MUICard
      data-cy={dataCy}
      style={{ margin: `${theme.spacing(2)}px`, width: `calc(100% - ${theme.spacing(4)}px)`, ...style }}
    >
      {cardHeader}
      <MUICardContent
        data-cy={getComposedDataCy(dataCy, SUBPARTS_MAP.content)}
        style={{ padding: `${theme.spacing(1)}px ${theme.spacing(2)}px` }}
      >
        {loading ? <MUISkeleton height={`${theme.spacing(16)}px`} /> : children}
      </MUICardContent>
      {!loading && (
        <Fragment>
          <MUICardActions
            disableSpacing
            style={{
              alignItems: "center",
              display: "flex",
              justifyContent: "space-between",
              padding: `${theme.spacing(1)}px ${theme.spacing(2)}px`,
            }}
          >
            {actions.length > 0 && (
              <div style={{ alignItems: "center", display: "flex" }}>
                {actions.map((action, index) => (
                  <div key={`card-action-${index}`} style={{ marginRight: `${theme.spacing(2)}px` }}>
                    {cloneElement(action)}{" "}
                  </div>
                ))}
              </div>
            )}
            {collapsible && (
              <IconButton
                dataCy={getComposedDataCy(dataCy, SUBPARTS_MAP.collapse)}
                icon={expanded ? Icons.up : Icons.down}
                onClick={() => setExpanded(!expanded)}
              />
            )}
          </MUICardActions>
          {collapsible && (
            <MUICollapse in={expanded} timeout="auto" unmountOnExit={unmountCollapsible}>
              <MUICardContent
                data-cy={getComposedDataCy(dataCy, SUBPARTS_MAP.collapsibleContent)}
                style={{ padding: `${theme.spacing(1)}px ${theme.spacing(2)}px` }}
              >
                {collapsible}
              </MUICardContent>
            </MUICollapse>
          )}
        </Fragment>
      )}
    </MUICard>
  );
};

export const CardWithProps = Card;

export default localized(Card, {
  dataCyShortcut: DATA_CY_SHORTCUT,
  localizableProps: LOCALIZABLE_PROPS,
});
