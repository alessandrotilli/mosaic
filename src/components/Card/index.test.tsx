import React from "react";
import renderer from "react-test-renderer";

import { ICard } from "../../types/Card";
import { getLocalizedTestable } from "../../utils/tests";

import Card, { DATA_CY_DEFAULT } from ".";
import { Icons } from "../../types/Icon";
import { MessageMock, mockedMessages, LocaleMock } from "../../utils/mocks/IntlProviderMock";
import IconButton from "../IconButton";
import Typography from "../Typography";

const defaultProps: ICard = {
  title: "Card",
};

const getCardTestable = (props?: ICard, dataCy = DATA_CY_DEFAULT) =>
  getLocalizedTestable(Card, { dataCy, domNode: "div", props: { ...defaultProps, ...props } });

describe("Card test suite:", () => {
  it("default", () => {
    const { element, wrapper } = getCardTestable();
    expect(wrapper).toHaveLength(1);
    const title = wrapper.find(`h2[data-cy='${DATA_CY_DEFAULT}-title']`);
    expect(title.text()).toEqual(defaultProps.title);

    const snapshotWrapper = renderer.create(element).toJSON();
    expect(snapshotWrapper).toMatchSnapshot();
  });

  it("dataCy", () => {
    const { element, wrapper } = getCardTestable({ ...defaultProps, dataCy: "custom" }, "custom");
    expect(wrapper).toHaveLength(1);

    const snapshotWrapper = renderer.create(element).toJSON();
    expect(snapshotWrapper).toMatchSnapshot();
  });

  it("localized", () => {
    const title = MessageMock.title;
    const props = { ...defaultProps, localized: true, title };
    const { element, wrapper } = getCardTestable({ ...props }, title);
    const titleWrapper = wrapper.find(`h2[data-cy='${title}-title']`);
    expect(titleWrapper.text()).toEqual(mockedMessages[LocaleMock.en][title]);

    const snapshotWrapper = renderer.create(element).toJSON();
    expect(snapshotWrapper).toMatchSnapshot();
  });

  it("actions", () => {
    const { element, wrapper } = getCardTestable({
      ...defaultProps,
      actions: [
        <IconButton dataCy="first-action" icon={Icons.account} onClick={() => {}} />,
        <IconButton dataCy="second-action" icon={Icons.account} onClick={() => {}} />,
      ],
    });
    const firstActionButton = wrapper.find("button[data-cy='first-action']");
    expect(firstActionButton).toHaveLength(1);
    const secondActionButton = wrapper.find("button[data-cy='second-action']");
    expect(secondActionButton).toHaveLength(1);

    const snapshotWrapper = renderer.create(element).toJSON();
    expect(snapshotWrapper).toMatchSnapshot();
  });

  it("collapsible", () => {
    const { element, wrapper } = getCardTestable({
      ...defaultProps,
      collapsible: <Typography dataCy="collapsible-typography">Collapsible</Typography>,
      unmountCollapsible: true,
    });
    const collapseButton = wrapper.find(`button[data-cy='${DATA_CY_DEFAULT}-collapse']`);
    collapseButton.simulate("click");

    const snapshotWrapper = renderer.create(element).toJSON();
    expect(snapshotWrapper).toMatchSnapshot();
  });

  it("icon", () => {
    const { element, wrapper } = getCardTestable({ ...defaultProps, icon: Icons.account });
    const icon = wrapper.find(`Icon[dataCy='${DATA_CY_DEFAULT}-avatar-icon']`);
    expect(icon.prop("name")).toEqual(Icons.account);

    const snapshotWrapper = renderer.create(element).toJSON();
    expect(snapshotWrapper).toMatchSnapshot();
  });

  it("loading", () => {
    const { element, wrapper } = getCardTestable({ ...defaultProps, loading: true });
    const placeholder = wrapper.find("span.MuiSkeleton-root");
    expect(placeholder).toHaveLength(3);

    const snapshotWrapper = renderer.create(element).toJSON();
    expect(snapshotWrapper).toMatchSnapshot();
  });

  it("subtitle", () => {
    const subtitle = "Subtitle";
    const { element, wrapper } = getCardTestable({ ...defaultProps, subtitle });
    const subtitleWrapper = wrapper.find(`span[data-cy='${DATA_CY_DEFAULT}-subtitle']`);
    expect(subtitleWrapper.text()).toEqual(subtitle);

    const snapshotWrapper = renderer.create(element).toJSON();
    expect(snapshotWrapper).toMatchSnapshot();
  });
});
