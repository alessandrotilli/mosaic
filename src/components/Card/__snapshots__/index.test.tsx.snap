// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card test suite: actions 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
  data-cy="card"
  style={
    {
      "margin": "16px",
      "width": "calc(100% - 32px)",
    }
  }
>
  <div
    className="MuiCardHeader-root css-185gdzj-MuiCardHeader-root"
  >
    <div
      className="MuiCardHeader-content css-1qbkelo-MuiCardHeader-content"
    >
      <h2
        className="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap css-1540tig-MuiTypography-root"
        data-cy="card-title"
      >
        Card
      </h2>
      <span
        className="MuiTypography-root MuiTypography-caption MuiTypography-noWrap css-840020-MuiTypography-root"
        data-cy="card-subtitle"
      />
    </div>
  </div>
  <div
    className="MuiCardContent-root css-46bh2p-MuiCardContent-root"
    data-cy="card-content"
    style={
      {
        "padding": "8px 16px",
      }
    }
  />
  <div
    className="MuiCardActions-root css-1rwjz6-MuiCardActions-root"
    style={
      {
        "alignItems": "center",
        "display": "flex",
        "justifyContent": "space-between",
        "padding": "8px 16px",
      }
    }
  >
    <div
      style={
        {
          "alignItems": "center",
          "display": "flex",
        }
      }
    >
      <div
        style={
          {
            "marginRight": "16px",
          }
        }
      >
        <button
          className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeMedium css-zylse7-MuiButtonBase-root-MuiIconButton-root"
          data-cy="first-action"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-cy="first-action-icon"
            data-testid="AccountCircleIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 4c1.93 0 3.5 1.57 3.5 3.5S13.93 13 12 13s-3.5-1.57-3.5-3.5S10.07 6 12 6zm0 14c-2.03 0-4.43-.82-6.14-2.88C7.55 15.8 9.68 15 12 15s4.45.8 6.14 2.12C16.43 19.18 14.03 20 12 20z"
            />
          </svg>
        </button>
         
      </div>
      <div
        style={
          {
            "marginRight": "16px",
          }
        }
      >
        <button
          className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeMedium css-zylse7-MuiButtonBase-root-MuiIconButton-root"
          data-cy="second-action"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-cy="second-action-icon"
            data-testid="AccountCircleIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 4c1.93 0 3.5 1.57 3.5 3.5S13.93 13 12 13s-3.5-1.57-3.5-3.5S10.07 6 12 6zm0 14c-2.03 0-4.43-.82-6.14-2.88C7.55 15.8 9.68 15 12 15s4.45.8 6.14 2.12C16.43 19.18 14.03 20 12 20z"
            />
          </svg>
        </button>
         
      </div>
    </div>
  </div>
</div>
`;

exports[`Card test suite: collapsible 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
  data-cy="card"
  style={
    {
      "margin": "16px",
      "width": "calc(100% - 32px)",
    }
  }
>
  <div
    className="MuiCardHeader-root css-185gdzj-MuiCardHeader-root"
  >
    <div
      className="MuiCardHeader-content css-1qbkelo-MuiCardHeader-content"
    >
      <h2
        className="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap css-1540tig-MuiTypography-root"
        data-cy="card-title"
      >
        Card
      </h2>
      <span
        className="MuiTypography-root MuiTypography-caption MuiTypography-noWrap css-840020-MuiTypography-root"
        data-cy="card-subtitle"
      />
    </div>
  </div>
  <div
    className="MuiCardContent-root css-46bh2p-MuiCardContent-root"
    data-cy="card-content"
    style={
      {
        "padding": "8px 16px",
      }
    }
  />
  <div
    className="MuiCardActions-root css-1rwjz6-MuiCardActions-root"
    style={
      {
        "alignItems": "center",
        "display": "flex",
        "justifyContent": "space-between",
        "padding": "8px 16px",
      }
    }
  >
    <button
      className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeMedium css-zylse7-MuiButtonBase-root-MuiIconButton-root"
      data-cy="card-collapse"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-cy="card-collapse-icon"
        data-testid="KeyboardArrowDownIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M7.41 8.59 12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
        />
      </svg>
      <span
        className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;

exports[`Card test suite: dataCy 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
  data-cy="custom"
  style={
    {
      "margin": "16px",
      "width": "calc(100% - 32px)",
    }
  }
>
  <div
    className="MuiCardHeader-root css-185gdzj-MuiCardHeader-root"
  >
    <div
      className="MuiCardHeader-content css-1qbkelo-MuiCardHeader-content"
    >
      <h2
        className="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap css-1540tig-MuiTypography-root"
        data-cy="custom-title"
      >
        Card
      </h2>
      <span
        className="MuiTypography-root MuiTypography-caption MuiTypography-noWrap css-840020-MuiTypography-root"
        data-cy="custom-subtitle"
      />
    </div>
  </div>
  <div
    className="MuiCardContent-root css-46bh2p-MuiCardContent-root"
    data-cy="custom-content"
    style={
      {
        "padding": "8px 16px",
      }
    }
  />
  <div
    className="MuiCardActions-root css-1rwjz6-MuiCardActions-root"
    style={
      {
        "alignItems": "center",
        "display": "flex",
        "justifyContent": "space-between",
        "padding": "8px 16px",
      }
    }
  />
</div>
`;

exports[`Card test suite: default 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
  data-cy="card"
  style={
    {
      "margin": "16px",
      "width": "calc(100% - 32px)",
    }
  }
>
  <div
    className="MuiCardHeader-root css-185gdzj-MuiCardHeader-root"
  >
    <div
      className="MuiCardHeader-content css-1qbkelo-MuiCardHeader-content"
    >
      <h2
        className="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap css-1540tig-MuiTypography-root"
        data-cy="card-title"
      >
        Card
      </h2>
      <span
        className="MuiTypography-root MuiTypography-caption MuiTypography-noWrap css-840020-MuiTypography-root"
        data-cy="card-subtitle"
      />
    </div>
  </div>
  <div
    className="MuiCardContent-root css-46bh2p-MuiCardContent-root"
    data-cy="card-content"
    style={
      {
        "padding": "8px 16px",
      }
    }
  />
  <div
    className="MuiCardActions-root css-1rwjz6-MuiCardActions-root"
    style={
      {
        "alignItems": "center",
        "display": "flex",
        "justifyContent": "space-between",
        "padding": "8px 16px",
      }
    }
  />
</div>
`;

exports[`Card test suite: icon 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
  data-cy="card"
  style={
    {
      "margin": "16px",
      "width": "calc(100% - 32px)",
    }
  }
>
  <div
    className="MuiCardHeader-root css-185gdzj-MuiCardHeader-root"
  >
    <div
      className="MuiCardHeader-avatar css-1ssile9-MuiCardHeader-avatar"
    >
      <div
        className="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-2s90m6-MuiAvatar-root"
        data-cy="card-avatar"
        style={
          {
            "backgroundColor": "#1976d2",
            "color": "#fff",
          }
        }
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-cy="card-avatar-icon"
          data-testid="AccountCircleIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 4c1.93 0 3.5 1.57 3.5 3.5S13.93 13 12 13s-3.5-1.57-3.5-3.5S10.07 6 12 6zm0 14c-2.03 0-4.43-.82-6.14-2.88C7.55 15.8 9.68 15 12 15s4.45.8 6.14 2.12C16.43 19.18 14.03 20 12 20z"
          />
        </svg>
      </div>
    </div>
    <div
      className="MuiCardHeader-content css-1qbkelo-MuiCardHeader-content"
    >
      <h2
        className="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap css-1540tig-MuiTypography-root"
        data-cy="card-title"
      >
        Card
      </h2>
      <span
        className="MuiTypography-root MuiTypography-caption MuiTypography-noWrap css-840020-MuiTypography-root"
        data-cy="card-subtitle"
      />
    </div>
  </div>
  <div
    className="MuiCardContent-root css-46bh2p-MuiCardContent-root"
    data-cy="card-content"
    style={
      {
        "padding": "8px 16px",
      }
    }
  />
  <div
    className="MuiCardActions-root css-1rwjz6-MuiCardActions-root"
    style={
      {
        "alignItems": "center",
        "display": "flex",
        "justifyContent": "space-between",
        "padding": "8px 16px",
      }
    }
  />
</div>
`;

exports[`Card test suite: loading 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
  data-cy="card"
  style={
    {
      "margin": "16px",
      "width": "calc(100% - 32px)",
    }
  }
>
  <div
    className="MuiCardHeader-root css-185gdzj-MuiCardHeader-root"
  >
    <div
      className="MuiCardHeader-content css-1qbkelo-MuiCardHeader-content"
    >
      <h2
        className="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap css-1540tig-MuiTypography-root"
        data-cy="card-title-loading"
      >
        <span
          className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1n7gpw1-MuiSkeleton-root"
          style={
            {
              "height": undefined,
              "width": undefined,
            }
          }
        />
      </h2>
      <span
        className="MuiTypography-root MuiTypography-caption MuiTypography-noWrap css-840020-MuiTypography-root"
        data-cy="card-subtitle-loading"
      >
        <span
          className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1n7gpw1-MuiSkeleton-root"
          style={
            {
              "height": undefined,
              "width": undefined,
            }
          }
        />
      </span>
    </div>
  </div>
  <div
    className="MuiCardContent-root css-46bh2p-MuiCardContent-root"
    data-cy="card-content"
    style={
      {
        "padding": "8px 16px",
      }
    }
  >
    <span
      className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1n7gpw1-MuiSkeleton-root"
      style={
        {
          "height": "128px",
          "width": undefined,
        }
      }
    />
  </div>
</div>
`;

exports[`Card test suite: localized 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
  data-cy="locale.title"
  style={
    {
      "margin": "16px",
      "width": "calc(100% - 32px)",
    }
  }
>
  <div
    className="MuiCardHeader-root css-185gdzj-MuiCardHeader-root"
  >
    <div
      className="MuiCardHeader-content css-1qbkelo-MuiCardHeader-content"
    >
      <h2
        className="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap css-1540tig-MuiTypography-root"
        data-cy="locale.title-title"
      >
        Title
      </h2>
      <span
        className="MuiTypography-root MuiTypography-caption MuiTypography-noWrap css-840020-MuiTypography-root"
        data-cy="locale.title-subtitle"
      />
    </div>
  </div>
  <div
    className="MuiCardContent-root css-46bh2p-MuiCardContent-root"
    data-cy="locale.title-content"
    style={
      {
        "padding": "8px 16px",
      }
    }
  />
  <div
    className="MuiCardActions-root css-1rwjz6-MuiCardActions-root"
    style={
      {
        "alignItems": "center",
        "display": "flex",
        "justifyContent": "space-between",
        "padding": "8px 16px",
      }
    }
  />
</div>
`;

exports[`Card test suite: subtitle 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
  data-cy="card"
  style={
    {
      "margin": "16px",
      "width": "calc(100% - 32px)",
    }
  }
>
  <div
    className="MuiCardHeader-root css-185gdzj-MuiCardHeader-root"
  >
    <div
      className="MuiCardHeader-content css-1qbkelo-MuiCardHeader-content"
    >
      <h2
        className="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap css-1540tig-MuiTypography-root"
        data-cy="card-title"
      >
        Card
      </h2>
      <span
        className="MuiTypography-root MuiTypography-caption MuiTypography-noWrap css-840020-MuiTypography-root"
        data-cy="card-subtitle"
      >
        Subtitle
      </span>
    </div>
  </div>
  <div
    className="MuiCardContent-root css-46bh2p-MuiCardContent-root"
    data-cy="card-content"
    style={
      {
        "padding": "8px 16px",
      }
    }
  />
  <div
    className="MuiCardActions-root css-1rwjz6-MuiCardActions-root"
    style={
      {
        "alignItems": "center",
        "display": "flex",
        "justifyContent": "space-between",
        "padding": "8px 16px",
      }
    }
  />
</div>
`;
